rust/lance-encoding/src/compression_config.rs

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+//! Configuration for compression parameters
+//!
+//! This module provides types for configuring compression strategies
+//! on a per-column or per-type basis using a parameter-driven approach.
+
+use std::collections::HashMap;
+
+use arrow::datatypes::DataType;
+
+/// Compression parameter configuration
+#[derive(Debug, Clone, PartialEq)]
+pub struct CompressionParams {
+    /// Column-level parameters: column name/pattern -> parameters
+    pub columns: HashMap<String, CompressionFieldParams>,
+
+    /// Type-level parameters: data type name -> parameters
+    pub types: HashMap<String, CompressionFieldParams>,
+}
+
+/// Field-level compression parameters
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct CompressionFieldParams {
+    /// RLE threshold (0.0-1.0)
+    /// When run_count < num_values * threshold, RLE will be used
+    pub rle_threshold: Option<f64>,
+
+    /// General compression scheme: "lz4", "zstd", "none"
+    pub compression: Option<String>,
+
+    /// Compression level (only for schemes that support it, e.g., zstd)
+    pub compression_level: Option<i32>,
+}
```

Comment:
In the old approach we used field metadata and so configuration parameters were in a string-string hashmap.  One thing we had considered at one point that seemed interesting to people was eventually getting to the point where new encodings could be added as plugins.

If we did want to do plugins then we could easily support new configuration parameters because it was just a string-string hashmap.

Can this approach also work with plugins?